packages/serverless/src/model/abstract/document.ts

```diff
@@ -102,146 +52,37 @@ export class ModelDocument extends ModelBasic<
-   * field types.
-   */
-  public static serializeDocument(
-    document: Record<string, TDocumentField>
-  ): Record<string, TContractSchemaFieldSerializable> {
-    return Object.values(document)
-      .map((field) => {
-        switch (field.kind) {
-          case 'UInt32':
-            return {
-              ...field,
-              value: new Long(field.value),
-            };
-          case 'Int64':
-            return {
-              ...field,
-              value: new Long(field.value),
-            };
-          case 'Bool':
-          case 'Sign':
-          case 'Character':
-          case 'PublicKey':
-          case 'PrivateKey':
-          case 'Signature':
-          case 'CircuitString':
-            return field;
-          case 'Field':
-          case 'UInt64':
-          case 'MerkleMapWitness':
-            throw new Error(
-              `Field type ${field.kind} is not yet supported in database`
-            );
-          default:
-            throw new Error(
-              `Unhandled field type, it is required that we handle all \
-possible field kinds explicitly to ensure correctness.`
-            );
-        }
-      })
-      .reduce(
-        (acc, field) => {
-          acc[field.name] = field;
-          return acc;
-        },
-        {} as Record<string, TContractSchemaFieldSerializable>
-      );
-  }
-
-  /** Deserializes a document by converting its fields to the appropriate
-   * document field types. */
-  public static deserializeDocument(
-    document: Record<string, TContractSchemaFieldSerializable>
-  ): Record<string, TDocumentField> {
-    return Object.values(document)
-      .map((field) => {
-        switch (field.kind) {
-          case 'UInt32':
-            return {
-              ...field,
-              value: field.value.toNumber(),
-            };
-          case 'Int64':
-            return {
-              ...field,
-              value: field.value.toBigInt(),
-            };
-          case 'Bool':
-          case 'Sign':
-          case 'Character':
-          case 'PublicKey':
-          case 'PrivateKey':
-          case 'Signature':
-          case 'CircuitString':
-            return field;
-          case 'Field':
-          case 'UInt64':
-          case 'MerkleMapWitness':
-            throw new Error(
-              `Field type ${field.kind} is not yet supported in database`
-            );
-          default:
-            throw new Error(
-              `Unhandled field type, it is required that we handle all \
-possible field kinds explicitly to ensure correctness.`
-            );
-        }
-      })
-      .reduce(
-        (acc, field) => {
-          acc[field.name] = field;
-          return acc;
-        },
-        {} as Record<string, TDocumentField>
-      );
-  }
-
   /** Construct a document with fields and insert it to the collection, marking
    * it as active */
   public async insertOneFromFields(
```

Comment:
fromFields is more specific and clear because it construct a new document with default values such as timestamps, which is not the same as inserting as is